sys/arm64/arm64/cheri_revoke_machdep.c

```diff
@@ -94,7 +94,9 @@ vm_do_cheri_revoke(int *res,
 		const uint8_t * __capability crshadow,
 		vm_cheri_revoke_test_fn ctp,
 		uintcap_t * __capability cutp,
-		uintcap_t cut)
+		uintcap_t cut,
+		vm_offset_t start,
+		vm_offset_t end)
```

Comment:
I'm not opposed to passing two offsets, but I wonder if it would be slightly better to pass one capability spanning the range, especially if we had a CInBounds instruction?  I guess that gets into alignment/representability.

Reply:
Hmm, that's in interesting idea. That would complicate the merging on quarantine since you could only merge entries where the result was representable so you'd probably end up wanting to examine the whole range of adjacent quarantined entries and produce a new, optimal set of cuts.

Reply:
One comment on adding these is that IIRC even only adding one pushes us into a stack spill on x86 so in the distant future we might want to pay some other costs to avoid that or ensure the use of the callback is optimized away.